content/en/docs/components/katib/user-guides/hp-tuning/llm-hp-optimization.md

```diff
@@ -0,0 +1,351 @@
++++
+title = "How to Optimize Hyperparameters of LLMs with Kubeflow"
+description = "API description of LLM hyperparameter optimization using Katib"
+weight = 20
++++
+
+{{% alert title="Warning" color="warning" %}}
+This feature is in **alpha** stage and the Kubeflow community is looking for your feedback. Please
+share your experience using the [#kubeflow-katib Slack channel](https://cloud-native.slack.com/archives/C073N7AS48P)
+or the [Kubeflow Katib GitHub](https://github.com/kubeflow/katib/issues).
+{{% /alert %}}
+
+This page describes how to implement Hyperparameter Optimization (HPO) using Python API that Katib supports and how to configure
+it.
+
+## Sections
+- [Prerequisites](#prerequisites)
+- [Load Model and Dataset](#load-model-and-dataset)
+- [Optimizing Hyperparameters of Large Language Models](#optimizing-hyperparameters-of-large-language-models)
+- [Example: Optimizing Hyperparameters of Llama-3.2 for Binary Classification on IMDB Dataset](#example-optimizing-hyperparameters-of-llama-32-for-binary-classification-on-imdb-dataset)
+
+## Prerequisites
+
+You need to install the following Katib components to run code in this guide:
+
+- Training Operator control plane – [install](/docs/components/training/installation/#installing-the-control-plane)
+- Katib control plane – [install](/docs/components/katib/installation/#installing-control-plane)
+- Katib Python SDK with LLM Hyperparameter Optimization Support – [install](/docs/components/katib/installation/#installing-python-sdk)
+
+**Note:** If you choose to define your own custom objective function and optimize parameters within it, distributed training is currently not supported. In this case, installing the Training Operator control plane is not necessary. For detailed instructions, please refer to [this guide](/docs/components/katib/getting-started).
+
+**Note:** Ensure that the Training Operator control plane is installed prior to the Katib control plane. This guarantees that the correct namespace labels are applied and enables the PyTorchJob CRD to be utilized in Katib.
+
+## Load Model and Dataset
+
+To optimize hyperparameters of a pre-trained model, it is essential to load the model and dataset from a provider. Currently, this can be done using external platforms like **Hugging Face** and **S3-compatible object storage** (e.g., Amazon S3) through the `storage_initializer` API from Kubeflow.
+
+### Hugging Face Integration
+
+The Hugging Face provider enables seamless integration of models and datasets for training and evaluation. You can import the necessary components for Hugging Face using the following code:
+
+```python
+from kubeflow.storage_initializer.hugging_face import (
+    HuggingFaceModelParams,
+    HuggingFaceDatasetParams,
+    HuggingFaceTrainerParams,
+)
+```
+
+#### HuggingFaceModelParams
```

Comment:
Can we move these sections to the Training Operator doc and cross-reference it from this doc ?
https://www.kubeflow.org/docs/components/trainer/legacy-v1/user-guides/fine-tuning/